docs/configure-gcp.md

```diff
@@ -0,0 +1,117 @@
+
+Create two projects from two different google accounts, one served as the service account's project and one as the tenant's project. 
+
+#### Service account Configuration
+
+Create a new Service account:
+
+1. Navigate to the Service account project
+2. Navigate to IAM & Admin on GCP.
+3. Click Service Accounts
+4. Click Create Service Account.
+5. Fill in the Service Account details.
+6. Click CREATE AND CONTINUE.
+7. Choose the permissions the provisioning app needs (TBC):
+
+Roles:
+ - Service Account User Role
+  
+Permissions:
+  - compute.instances.insert
+
+8. Click CONTINUE
+9.  Click DONE 
+10. Copy the service account's Email
+
+
+### Tenant account
+
+This is a setup in the account in which the service shall deploy the actual instances.
+
+#### Tenant Configuration
+
+1. Navigate to the Tenant project
+2. Navigate to IAM & Admin on GCP.
+3. Click IAM.
+4. Click +ADD.
+5. Fill in the Service Account Email under New principals.
+6. Choose the permissions the provisioning app needs (TBC): 
+Roles:
+ - Service Account User Role
+  
+Permissions:
+  - compute.instances.insert
+
+7. Click SAVE.
+
+#### Authenticating as the service account
+
+1. In the Google Cloud console, go to the Service accounts page.
+2. Select the service account project 
+3. On the Service accounts page, click the email address of the service account that you want to create a key for.
+4. Click the Keys tab.
+5. Click the Add key drop-down menu, then select Create new key.
+6. Select JSON as the Key type and click Create.
+Clicking Create downloads a service account key file. 
+
+
+The downloaded key has the following format, where PRIVATE_KEY is the private portion of the public/private key pair:
+```json
+{
+  "type": "service_account",
+  "project_id": "PROJECT_ID",
+  "private_key_id": "KEY_ID",
+  "private_key": "-----BEGIN PRIVATE KEY-----\nPRIVATE_KEY\n-----END PRIVATE KEY-----\n",
+  "client_email": "SERVICE_ACCOUNT_EMAIL",
+  "client_id": "CLIENT_ID",
+  "auth_uri": "https://accounts.google.com/o/oauth2/auth",
+  "token_uri": "https://accounts.google.com/o/oauth2/token",
+  "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
+  "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/SERVICE_ACCOUNT_EMAIL"
+}
+```
+7. Set the GOOGLE_APPLICATION_CREDENTIALS environment variable to the downloaded json:
+
+export GOOGLE_APPLICATION_CREDENTIALS="{
+  "type": "service_account",
+  "project_id": "PROJECT_ID",
+  "private_key_id": "KEY_ID",
+  "private_key": "-----BEGIN PRIVATE KEY-----\nPRIVATE_KEY\n-----END PRIVATE KEY-----\n",
+  "client_email": "SERVICE_ACCOUNT_EMAIL",
+  "client_id": "CLIENT_ID",
+  "auth_uri": "https://accounts.google.com/o/oauth2/auth",
+  "token_uri": "https://accounts.google.com/o/oauth2/token",
+  "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
+  "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/SERVICE_ACCOUNT_EMAIL"
+}"
```

Comment:
This is not a valid shell syntax, this will fail.

The JSON format is very unfortunate, I wish we could simply have a configurable option of a filename that would be then loaded from the filesystem. But we run in a container.

Before we commit to a solution, let's investigate how clowder is capable in this regard. @ezr-ondrej @mshriver do you happen to know if clowder/sre even supports multiline values?

If not, we can simply break this JSON into smaller values:

* project id
* key id
* client email
* client id
* key encoded as base64 without newlines

Then our app can create the JSON on the fly.

Reply:
Oh another idea, we can simply remove all newlines, but still this has a problem of many quote characters which need to be escaped in shell or even in YAML. That's why I think it's dangerous and base64 might be better fit.

I tried to compress it but it's not much shorter.

Reply:
Can you rewrite this to use base64 encoded string? Simply instruct the reader to do:

```
cat file.json | base64
```

The output then must be used for the environmental variable which is now named GCP_JSON (or in local.yaml). This is indeed in my PR, so it hasn't been merged yet.

Reply:
How do you want to write it down? Is that enough? 
7. Set the GCP_JSON environment variable to the downloaded JSON in base64:
   ```shell
   cat creds.json | base 64
  ```

Reply:
Yeah, the command is called `base64` not with space.